sim/pokemon.ts

```diff
@@ -517,7 +516,9 @@ export class Pokemon {
 		const health = this.getHealth();
 		let details = this.details;
 		if (this.illusion) {
-			details = this.illusion.getUpdatedDetails(this.battle.ruleTable.has('illusionlevelmod') ? this.level : undefined);
+			details = this.illusion.getUpdatedDetails(
+				this.battle.ruleTable.has('illusionlevelmod') ? this.illusion.level : undefined
+			);
```

Comment:
I now think illusion level mod is permanently active since the default value for `this.illusion.getUpdateDetails` is already `this.illusion.level`...